spec/lib/audit/catalog_to_moab_class_spec.rb

```diff
@@ -6,31 +6,41 @@
   let(:storage_dir) { 'spec/fixtures/storage_root01/moab_storage_trunk' }
   let(:limit) { Settings.c2m_sql_limit }
 
-  context '.check_version_on_dir_of_batch' do
+  context '.check_version_on_dir' do
     include_context 'fixture moabs in db'
-    let(:subject) { described_class.check_version_on_dir_of_batch(last_checked_version_b4_date, storage_dir, limit) }
+    let(:subject) { described_class.check_version_on_dir(last_checked_version_b4_date, storage_dir, limit) }
 
-    it 'creates an instance and calls #check_catalog_version' do
-      c2m_mock = instance_double(described_class)
-      allow(described_class).to receive(:new).and_return(c2m_mock)
-      expect(c2m_mock).to receive(:check_catalog_version).exactly(3).times
-      described_class.check_version_on_dir_of_batch(last_checked_version_b4_date, storage_dir, limit)
-    end
-  end
+    context 'when there are PreservedCopies to check' do
+      let(:c2m_mock) { instance_double(described_class) }
 
-  context ".check_version_on_dir" do
-    include_context 'fixture moabs in db'
-    let(:subject) { described_class.check_version_on_dir(last_checked_version_b4_date, storage_dir) }
+      it 'creates an instance and calls #check_catalog_version within the limit' do
+        allow(described_class).to receive(:new).and_return(c2m_mock)
+        expect(c2m_mock).to receive(:check_catalog_version).exactly(3).times
+        described_class.check_version_on_dir(last_checked_version_b4_date, storage_dir, limit)
+      end
 
-    it "calls check_version_on_dir_of_batch when there are objects to audit" do
-      expect(described_class).to receive(:check_version_on_dir_of_batch).exactly(1).times
-      subject
+      it 'creates an instance and calls #check_catalog_version on everything in batches' do
+        # we use 2 separate loops as we have to do all of our .new calls on CatalogToMoab before the allowance
+        pcs_from_scope = PreservedCopy.least_recent_version_audit(last_checked_version_b4_date, storage_dir)
+        c2m_list = pcs_from_scope.map do |pc|
+          described_class.new(pc, storage_dir)
+        end
+        c2m_list.each do |c2m|
+          allow(described_class).to receive(:new).with(c2m.preserved_copy, storage_dir).and_return(c2m)
+          # we use and_call_original to get the audit timestamps updated
```

Comment:
this comment, to me, is overexplaining -- it's explaining what `.and_call_original` does, which is standard rspec.  I'm open to other opinions.

Reply:
yeah, this is another thing that bit us (our test was failing because we weren't actually updating audit timestamps, so each batch just kept trying to grab the same first couple pres copies).

but this one's maybe more obvious than the two separate loops thing?

Reply:
I hadn't run into `.and_call_original` before, so from that point of view, I feel like it's helpful, but since it's standard rspec, I'll delete it, and if we can discuss it further, if necessary.